cmd/govim/quickfix.go

```diff
@@ -23,11 +23,17 @@ func (q quickfixEntry) equalModuloBuffer(q2 quickfixEntry) bool {
 }
 
 func (v *vimstate) quickfixDiagnostics(flags govim.CommandFlags, args ...string) error {
+	wasNotDiagnostics := !v.quickfixIsDiagnostics
 	v.quickfixIsDiagnostics = true
-	return v.updateQuickfix(true)
+	return v.updateQuickfixWithDiagnostics(true, wasNotDiagnostics)
 }
 
-func (v *vimstate) updateQuickfix(force bool) error {
+// updateQuickfixWithDiagnostics updates Vim's quickfix window with the current
+// diagnostics(), respecting config settings that are overridden by force.  The
+// rather clumsily named wasPrevNotDiagnostics can be set to true if it is
+// known to the caller that the quickfix window was previously (to this call)
+// being used for something other than diagnostics, e.g. references.
+func (v *vimstate) updateQuickfixWithDiagnostics(force bool, wasPrevNotDiagnostics bool) error {
```

Comment:
Agree that it's a bit weird name. Would it make sense to name it something in lines of `resetIndex` or `resetSelection` instead? Even though it has to be renamed when the logic is improved to stash last selected diagnostic.

Reply:
Yeh I think I'm tempted to leave it named badly for now. At least the comments are clear. This will need to be refactored as you say when this gets fixed properly.